packages/wrangler/src/type-generation/runtime/log-runtime-types-message.ts

```diff
@@ -8,53 +9,60 @@ import { logger } from "../../logger";
 export function logRuntimeTypesMessage(
 	outFile: string,
 	tsconfigTypes: string[],
-	isNodeCompat = false,
-	configPath: string | undefined
+	isNodeCompat = false
 ) {
 	const isWorkersTypesInstalled = tsconfigTypes.find((type) =>
 		type.startsWith("@cloudflare/workers-types")
 	);
 	const isNodeTypesInstalled = tsconfigTypes.find((type) => type === "node");
 	const updatedTypesString = buildUpdatedTypesString(tsconfigTypes, outFile);
-
-	logger.info(`✨ Runtime types written to ${outFile}`);
-
+	const maybeHasOldRuntimeFile = existsSync("./.wrangler/types/runtime.d.ts");
+	if (maybeHasOldRuntimeFile) {
+		logAction("Remove the old runtime.d.ts file");
+		logger.log(
+			chalk.dim(
+				"`wrangler types` now outputs runtime and Env types in one file.\nYou can now delete the ./.wrangler/types/runtime.d.ts and update your tsconfig.json`"
+			)
+		);
+		logger.log("");
+	}
+	if (isWorkersTypesInstalled) {
+		logAction(
+			"Migrate from @cloudflare/workers-types to generated runtime types"
+		);
+		logger.log(
+			chalk.dim(
+				"`wrangler types` now generates runtime types and supersedes @cloudflare/workers-types.\nYou should now uninstall @cloudflare/workers-types and remove it from your tsconfig.json."
+			)
+		);
+		logger.log("");
+	}
 	if (updatedTypesString) {
-		logger.info(dedent`
-			📣 Add the generated types to the types array in your tsconfig.json:
-
+		logAction("Update your tsconfig.json to include the generated types");
+		logger.log(
+			chalk.dim(dedent`
 				{
 					"compilerOptions": {
-						...
 						"types": ${updatedTypesString}
-						...
 					}
 				}
-
-		`);
-	} else if (isWorkersTypesInstalled) {
-		logger.info(dedent`
-			📣 Replace the existing "@cloudflare/workers-types" entry with the generated types path:
-				{
-					"compilerOptions": {
-						...
-						"types": ${updatedTypesString}
-						...
-					}
-				}
-
-		`);
-	}
-	if (isWorkersTypesInstalled) {
-		logger.info('📣 You can now uninstall "@cloudflare/workers-types".');
+		`)
+		);
+		logger.log("");
 	}
 	if (isNodeCompat && !isNodeTypesInstalled) {
-		logger.info(
-			'📣 Since you have Node.js compatibility mode enabled, you should consider adding Node.js for TypeScript by running "npm i --save-dev @types/node@20.8.3". Please see the docs for more details: https://developers.cloudflare.com/workers/languages/typescript/#transitive-loading-of-typesnode-overrides-cloudflareworkers-types'
+		logAction("Install types@node");
+		logger.log(
+			chalk.dim(
+				`Since you have the \`nodejs_compat\` flag, you should install Node.js types by running "npm i --save-dev @types/node${isWorkersTypesInstalled ? '@20.8.3".\nFor more details: https://developers.cloudflare.com/workers/languages/typescript/#known-issues' : '".'}`
+			)
```

Comment:
Something like the following might improve readibility?

```suggestion
				`Since you have the \`nodejs_compat\` flag, you should install Node.js types by running "npm i --save-dev ${isWorkersTypesInstalled ? "@types/node@20.8.3.\nFor more details: https://developers.cloudflare.com/workers/languages/typescript/#known-issues' : '@types/node.'}`
			)
```